model/src/application/interaction/application_command_autocomplete/mod.rs

```diff
@@ -63,6 +63,30 @@ pub struct ApplicationCommandAutocomplete {
     pub user: Option<User>,
 }
 
+impl ApplicationCommandAutocomplete {
+    /// ID of the user that invoked the interaction.
+    ///
+    /// This will first check for the [`member`]'s
+    /// [`user`][`PartialMember::user`]'s ID and, if not present, then check the
+    /// [`user`]'s ID.
+    ///
+    /// [`member`]: Self::member
+    /// [`user`]: Self::user
+    pub const fn author_id(&self) -> Option<Id<UserMarker>> {
+        if let Some(member) = &self.member {
+            if let Some(user) = &member.user {
+                return Some(user.id);
+            }
+        }
+
+        if let Some(user) = &self.user {
+            return Some(user.id);
+        }
+
+        None
+    }
```

Comment:
We've got some duplication here, so we can abstract this out to the `interaction` module, something like:

```rust
const fn author_id(user: Option<&User>, member: Option<&Member>) -> Option<Id<UserMarker>>;
```